### PR TITLE
Feature/integration tests yield token rebased

### PR DESCRIPTION
**Pull request by danoctavian**
## Context

Solves issue https://github.com/NexusMutual/smart-contracts/issues/326

## Changes proposed in this pull request

Add yield token incidents tests for executing redeem against DAI and ETH pegged tokens.

Add test for buy Cover.

## Test plan

This PR is adding integration tests, not new production code.

## Checklist

- [x] Rebased the base branch
- [x] Attached corresponding Github issue
- [x] Prefixed the name with the type of change (i.e. feat, chore, test)
- [x] Performed a self-review of my own code
- [x] Followed the style guidelines of this project
- [ ] Made corresponding changes to the documentation
- [x] Didn't generate new warnings
- [x] Didn't generate failures on existing tests
- [x] Added tests that prove my fix is effective or that my feature works

## Review

When reviewing a PR, please indicate intention in comments using the following emojis:
* :cake: = Nice to have but not essential.
* :bulb: = Suggestion or a comment based on personal opinion
* :hammer: = I believe this should be changed.
* :thinking: = I don’t understand something, do you mind giving me more context?
* :rocket: = Feedback

### CODE REVIEW

File: test/integration/Cover/buyCover.js [hunk 1]
```diff
@@ -0,0 +1,174 @@
+const { ethers } = require('hardhat');
```
Comment (danoctavian) [inline test/integration/Cover/buyCover.js:1]: just realized this file is not enabled in the test run. will fix.

**Comment by danoctavian** [comment]
failing because of the cowswap run time issue on circle. will re run.

Also another merge issue appeared.

**Review by roxdanila — CHANGES_REQUESTED** [review]
back to you to fix the conflicts generated by the truffle -> ethers PR

File: test/integration/setup.js [hunk 2]
```diff
@@ -302,14 +308,14 @@ async function setup() {
       },
       {
         productType: 2, // Yield Token Cover
-        ytcUnderlyingAsset: '0x0000000000000000000000000000000000000001',
+        ytcUnderlyingAsset: ybETH.address,
```
Comment (shark0der) [inline test/integration/setup.js:311]: I don't think this is the correct asset to be used here. It should be the underlying asset, not the covered one.

File: test/integration/setup.js [hunk 3]
```diff
@@ -302,14 +308,14 @@ async function setup() {
...
         coverAssets: 0b01, // ETH
         initialPriceRatio: 100,
         capacityReductionRatio: 0,
       },
       {
         productType: 2, // Yield Token Cover
-        ytcUnderlyingAsset: '0x0000000000000000000000000000000000000002',
+        ytcUnderlyingAsset: ybDAI.address,
```
Comment (shark0der) [inline test/integration/setup.js:318]: Same here.

File: test/integration/setup.js (hunk 2 above)
Comment (danoctavian) [reply test/integration/setup.js:311]: It is right, it's the covered asset.

There was a rename to this new name

https://github.com/NexusMutual/smart-contracts/blob/490c3a93cd888b6d34083be27df0fdd6ad860e12/contracts/modules/assessment/YieldTokenIncidents.sol#L294

File: test/integration/setup.js (hunk 3 above)
Comment (danoctavian) [reply test/integration/setup.js:318]: same https://github.com/NexusMutual/smart-contracts/blob/490c3a93cd888b6d34083be27df0fdd6ad860e12/contracts/modules/assessment/YieldTokenIncidents.sol#L294

File: test/integration/YieldTokenIncidents/submitClaim.js [hunk 4]
```diff
@@ -3,8 +3,9 @@ const {
   constants: { ZERO_ADDRESS },
 } = require('@openzeppelin/test-helpers');
 const { expect } = require('chai');
+const { AddressZero } = ethers.constants;
```
Comment (shark0der) [inline test/integration/YieldTokenIncidents/submitClaim.js:6]: `AddressZero` was imported but oz is still in use.

File: test/integration/YieldTokenIncidents/submitClaim.js [hunk 5]
```diff
@@ -17,59 +18,78 @@ const setTime = async timestamp => {
...
+    // Stake to open up capacity
+    await stake({ stakingPool: stakingPool0, staker: staker1, productId, period, gracePeriod });
 
+    // Buy Cover
     const expectedPremium = amount
       .mul(BigNumber.from(DEFAULT_PRODUCT_INITIALIZATION[0].targetPrice))
       .div(BigNumber.from(priceDenominator));
 
-    await stakingPool0.setTargetWeight(productId, 10);
+    const expectedMinCoveredAmount = amount.sub(COVERED_AMOUNT_PRECISION_ERROR_TOLERANCE);
```
Comment (shark0der) [inline test/integration/YieldTokenIncidents/submitClaim.js:77]: Is this something undeterministic that we cannot get precisely in js?

Comment (danoctavian) [reply test/integration/YieldTokenIncidents/submitClaim.js:77]: It's a precision error range. the precision error is dust, just 20 wei

File: test/integration/YieldTokenIncidents/submitClaim.js (hunk 4 above)
Comment (danoctavian) [reply test/integration/YieldTokenIncidents/submitClaim.js:3]: removed

**Pull request merged**